src/app/games/lasca/lasca.component.ts

```diff
@@ -0,0 +1,344 @@
+            }
+        }
+        // Continuing to capture
+        return this.capture(clicked);
+    }
+    private async capture(clicked: Coord): Promise<MGPValidation> {
+        const numberOfClicks: number = this.currentMoveClicks.length;
+        const lastCoord: Coord = this.currentMoveClicks[numberOfClicks - 1];
+        const delta: Coord = lastCoord.getVectorToward(clicked);
+        if (Math.abs(delta.x) === 2 && Math.abs(delta.y) === 2) {
+            const captured: Coord = new Coord(lastCoord.x + (delta.x / 2), lastCoord.y + (delta.y / 2));
+            this.capturedCoords.push(captured);
+            this.currentMoveClicks.push(clicked);
+            const currentMove: LascaMove = LascaMove.fromCapture(this.currentMoveClicks).get();
+            if (this.legalMoves.some((capture: LascaMove) => capture.isPrefix(currentMove))) {
+                return this.applyPartialCapture();
+            } else {
+                return this.chooseMove(currentMove, this.getState());
+            }
+        } else {
+            return this.cancelMove(LascaMoveFailure.CAPTURE_STEPS_MUST_BE_DOUBLE_DIAGONAL());
+        }
+    }
+    private applyPartialCapture(): MGPValidation {
+        let partialState: LascaState = this.getState().remove(this.currentMoveClicks[0]);
+        let movingStack: LascaSpace = this.getState().getPieceAt(this.currentMoveClicks[0]);
+        for (const captured of this.capturedCoords) {
+            const previousSpace: LascaSpace = this.getState().getPieceAt(captured);
+            const capturedCommander: LascaPiece = previousSpace.getCommander();
+            const commandedStack: LascaSpace = previousSpace.getPiecesUnderCommander();
+            partialState = partialState.set(captured, commandedStack);
+            movingStack = movingStack.capturePiece(capturedCommander);
+        }
+        const landing: Coord = this.currentMoveClicks[this.currentMoveClicks.length - 1];
+        partialState = partialState.set(landing, movingStack);
+        this.updateAdaptedBoardFrom(partialState);
+        this.rotateAdaptedBoardIfNeeded();
+        return MGPValidation.SUCCESS;
+    }
+    private async trySelectingPiece(clicked: Coord): Promise<MGPValidation> {
+        const clickedSpace: LascaSpace = this.getState().getPieceAt(clicked);
+        if (clickedSpace.isEmpty()) {
+            return this.cancelMove(RulesFailure.MUST_CHOOSE_OWN_PIECE_NOT_EMPTY());
+        } else {
+            return this.selectPiece(clicked);
+        }
+    }
+    private selectPiece(coord: Coord): MGPValidation {
+        this.capturedCoords = [];
+        if (this.legalMoves.some((move: LascaMove) => move.getStartingCoord().equals(coord))) {
+            this.currentMoveClicks = [coord];
+            this.hideLastMove();
+            return MGPValidation.SUCCESS;
+        } else {
+            return this.cancelMove(LascaComponentFailure.THIS_PIECE_CANNOT_MOVE());
+            // TODOTODO: QUAND TU CAPTURE DEUX RFOIS LA PILE GRANDI EH
```

Comment:
eh

Reply:
haha, j'avais noté ça quand le Set enlevais les doublon (et donc les piles dépassaient """jamais""" 2 pièces :D (mais elles auraient pu))